### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by GegoSK** [sphanley/Duolinguist#1]
Hello,
after many months using your integration, for first time it is not working. 

Error: Config entry 'GegoSK' for duolingo integration not ready yet: None; Retrying in background

Hope you look at it. 

Thank you for integration!

**Comment by sphanley** [comment]
Thanks for letting me know! I just tried a reload of my integration and now I'm seeing the same thing. I'll look into it when I have a chance.

**Comment by sphanley** [comment]
Unfortunately, this issue seems to occur when just using the [API caller](https://github.com/KartikTalwar/Duolingo) I'm dependent on. As best I can tell, Duolingo may have changed their login mechanism - the calls being made by that library no longer seem sufficient to log in. Unless someone else figures out how to headlessly auth with Duolingo, it's unlikely that this is going to be something I can resolve. Apologies, and thanks for using my little project!

**Comment by dontbug** [comment]
Thanks for your package. I also use and also noticed that it does not work for the last 2-3 days.

**Comment by dontbug** [comment]
Yes, they may have changed their authorization procedure. Even in their library, people write about the authorization error.
https://github.com/KartikTalwar/Duolingo/issues/128

**Comment by GegoSK** [comment]
Someone published a solution in link, which mentioned @dontbug. I do not know if it help you @sphanley.

**Comment by drbartek** [comment]
Any chance of an update? Children are out of control ;-)

**Comment by jellewie** [comment]
I would love to see this work

**Comment by wacher74** [comment]
I tried to add this integration to my HA, but I got this message
Config flow could not be loaded: {"message":"Invalid handler specified"}
Home Assistant 2023.7.3

**Comment by TimCinel** [comment]
I've got this integration working again using [tier61wro's fix](https://github.com/KartikTalwar/Duolingo/issues/128#issuecomment-1642372520).

Raised a [PR](https://github.com/sphanley/Duolinguist/pull/2) in case we want to merge it. The fix may not work forever, but it's an improvement on the current state...

**Comment by sphanley** [comment]
Hi everyone! Thanks for all the discussion, apologies for taking so long to chime in. Unfortunately, at this point, I'm not really inclined to make any updates to this project. It's something I mostly developed as a way to learn about making a custom integration, and I don't feel great about continuing to try and circumvent Duolingo's API access controls given that they seem to have intentionally tried to restrict unapproved access. Obviously folks can fork this project and do with it what they will, but I'm probably going to archive my copy of it, unless Duolingo eventually releases a sanctioned way to consume their APIs.
